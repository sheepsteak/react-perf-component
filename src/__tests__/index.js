import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import perf from '../';

describe('perf higher-order component', () => {
  let ReactPerf;

  beforeEach(() => {
    jest.resetModules();
    ReactPerf = jest.genMockFromModule('react-addons-perf');
    jest.setMock('react-addons-perf', ReactPerf);
  });

  it('wraps the passed in component when not in production', () => {
    const MyComp = () => <div className="my-comp" />;
    const MyCompPerf = perf(MyComp);

    const result = mount(<MyCompPerf testProp="1" />);

    expect(toJson(result)).toMatchSnapshot();
  });

  it('does not wrap the passed in component when in production', () => {
    const MyComp = () => <div className="my-comp" />;
    const NODE_ENV = process.env.NODE_ENV;
    process.env.NODE_ENV = 'production';
    const MyCompPerf = perf(MyComp);
    process.env.NODE_ENV = NODE_ENV;

    const result = mount(<MyCompPerf testProp="1" />);

    expect(toJson(result)).toMatchSnapshot();
  });

  it('starts ReactPerf on mount', () => {
    const MyComp = () => <div className="my-comp" />;
    const MyCompPerf = perf(MyComp);

    mount(<MyCompPerf />);

    expect(ReactPerf.start).toHaveBeenCalled();
  });

  it('stops ReactPerf on unmount', () => {
    const MyComp = () => <div className="my-comp" />;
    const MyCompPerf = perf(MyComp);

    const result = mount(<MyCompPerf />);
    result.unmount();

    expect(ReactPerf.stop).toHaveBeenCalled();
  });

  it('stops, collects results and starts again in update', () => {
    const MyComp = () => <div className="my-comp" />;
    const MyCompPerf = perf(MyComp);

    const result = mount(<MyCompPerf />);
    result.setProps();

    expect(ReactPerf.stop).toHaveBeenCalled();
    expect(ReactPerf.getLastMeasurements).toHaveBeenCalled();
    expect(ReactPerf.printWasted).toHaveBeenCalled();
    expect(ReactPerf.start).toHaveBeenCalled();
  });
});
