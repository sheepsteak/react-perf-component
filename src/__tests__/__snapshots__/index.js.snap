exports[`perf higher-order component does not wrap the passed in component when in production 1`] = `
<MyComp
  testProp="1">
  <div
    className="my-comp" />
</MyComp>
`;

exports[`perf higher-order component wraps the passed in component when not in production 1`] = `
<Perf
  testProp="1">
  <MyComp
    testProp="1">
    <div
      className="my-comp" />
  </MyComp>
</Perf>
`;
